config.go

```diff
@@ -557,33 +560,61 @@ func loadConfig() (*config, []string, error) {
 
 	// Ensure the admin password is set.
 	if cfg.AdminPass == "" {
-		str := "%s: the adminpass option is not set"
-		err := fmt.Errorf(str, funcName)
+		err := fmt.Errorf("the adminpass option is not set")
+		fmt.Fprintln(os.Stderr, err)
+		fmt.Fprintln(os.Stderr, usageMessage)
 		return nil, nil, err
 	}
 
 	// Ensure the dcrd rpc username is set.
 	if cfg.RPCUser == "" {
-		str := "%s: the rpcuser option is not set"
-		err := fmt.Errorf(str, funcName)
+		err := fmt.Errorf("the rpcuser option is not set")
+		fmt.Fprintln(os.Stderr, err)
+		fmt.Fprintln(os.Stderr, usageMessage)
 		return nil, nil, err
 	}
 
 	// Ensure the dcrd rpc password is set.
 	if cfg.RPCPass == "" {
-		str := "%s: the rpcpass option is not set"
-		err := fmt.Errorf(str, funcName)
+		err := fmt.Errorf("the rpcpass option is not set")
+		fmt.Fprintln(os.Stderr, err)
+		fmt.Fprintln(os.Stderr, usageMessage)
 		return nil, nil, err
 	}
 
 	// Create the data directory.
 	err = os.MkdirAll(cfg.DataDir, 0700)
 	if err != nil {
-		str := "%s: failed to create data directory: %v"
-		err := fmt.Errorf(str, funcName, err)
+		str := "%s: unable to create data directory (%s): %v"
```

Comment:
`funcName` is used in a handful of places, but not all. Any reason? I guess we could remove it completely

Reply:
I think it makes sense to keep it for config related errors resulting from creating directories, generating certificates etc which are not directly related to the config process.Errors related to config parsing and parameter constraints can do without the func name label since those indicate an app-wide config issue that's why `funcName` is used only in some places.